lib/acme/client/certificate_request.rb

```diff
@@ -79,7 +79,7 @@ def validate_subject_common_name
 
   def generate
     OpenSSL::X509::Request.new.tap do |csr|
-      csr.public_key = @private_key.public_key
```

Comment:
`OpenSSL::PKey::EC` is a private key type. Unfortunaltely, OpenSSL::X509::Request does not accept EC public key (`OpenSSL::PKey::EC::Point` class) so we have to use private key here.
This is a known limitation for EC use with ruby.